src/bosh-director/lib/bosh/director/dns/dns_version_converger.rb

```diff
@@ -3,9 +3,9 @@ class DnsVersionConverger
 
     ONLY_OUT_OF_DATE_SELECTOR = lambda do |current_version, logger|
       logger.info('Selected strategy: ONLY_OUT_OF_DATE_SELECTOR')
-      Models::Instance.inner_join(:vms, Sequel.qualify('vms', 'instance_id') => Sequel.qualify('instances', 'id'))
+      Config.db[:instances].inner_join(:vms, Sequel.qualify('vms', 'instance_id') => Sequel.qualify('instances', 'id'))
```

Comment:
I wouldn't use an inner_join, since ruby sequel makes a magic foreign key variable called: instance.vms: https://github.com/cloudfoundry/bosh/blob/master/src/bosh-director/lib/bosh/director/models/instance.rb#L9

If you have a given instance, you can just say: `instance.vms.each` to iterate through the matching vms for the instance, or `instance.vms.all` for an array of the vms.  And it should be super quick.  Alternately maybe an index would make the query resolve quicker if that isn't an option.

From the snippet I can't tell if that would help or not though.

Reply:
Not fully understanding the situation all the way, It looks to me like you want to use eager here 
https://sequel.jeremyevans.net/rdoc/classes/Sequel/Model/Associations/DatasetMethods.html#method-i-eager

(We it for similar reasons in some complicated lookups in cloud controller, like https://github.com/cloudfoundry/cloud_controller_ng/blob/63fa0c5815ab536aaa2cb5baed13452ca14ced6c/lib/cloud_controller/copilot/sync.rb#L70)

I'll try to take a closer look at the data model here if you think this is a useful direction.

Reply:
Thanks @charleshansen for the suggestion. I've added a second comment above where the change is maybe a bit clearer. I'll try to switch to `eager_graph` (I think that's necessary because we're filtering on `instance.vm.active`?) and see what happens.